grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/RedisTaskQueue.java

```diff
@@ -105,25 +113,36 @@ void putJob(Task job) {
     }
 
     void runInFlightProcessor() {
-        new Timer().scheduleAtFixedRate(new RedisInflightTaskConsumer(jedisPool, Duration.ofSeconds(600), config, QUEUE_NAME), new Date(), 10000);
+        new Timer().scheduleAtFixedRate(new RedisInflightTaskConsumer(jedisPool, Duration.ofSeconds(
+                processingDelay), config, QUEUE_NAME), new Date(), 2000);
     }
 
     void subscribe(
             RedisTaskManager redisTaskManager,
-            ExecutorService consumerExecutor,
             EngineID engineId,
-            GraknEngineConfig config,
-            EngineGraknGraphFactory factory) {
+            GraknEngineConfig engineConfig,
+            EngineGraknGraphFactory factory,
+            int poolSize) {
         LOG.info("Subscribing worker to jobs in queue {}", QUEUE_NAME);
-        Worker worker = getJobSubscriber();
+        // sync to avoid close while starting
+        synchronized(this) {
+            this.workerPool = new WorkerPool(() -> getWorker(redisTaskManager, engineId, engineConfig, factory), poolSize);
```

Comment:
Maybe in the future it's worth creating a central handles for all the thread related pools we make? I imagine us creating quite a lot.